prdoc/pr_6220.prdoc

```diff
@@ -0,0 +1,10 @@
+title: Improve `CheckMetadataHash` transaction extension weight and logic
```

Comment:
`title` needs to be updated, to reference the actual fix.

Reply:
:see_no_evil: thanks